pkg/planner/contextimpl/impl.go

```diff
@@ -28,13 +29,25 @@ var _ context.PlanContext = struct {
 
 // PlanCtxExtendedImpl provides extended method for session context to implement `PlanContext`
 type PlanCtxExtendedImpl struct {
-	sctx    sessionctx.Context
-	exprCtx exprctx.BuildContext
+	sctx                   sessionctx.Context
+	nullRejectCheckExprCtx *exprctx.NullRejectCheckExprContext
 }
 
 // NewPlanCtxExtendedImpl creates a new PlanCtxExtendedImpl.
 func NewPlanCtxExtendedImpl(sctx sessionctx.Context) *PlanCtxExtendedImpl {
-	return &PlanCtxExtendedImpl{sctx: sctx}
+	return &PlanCtxExtendedImpl{
+		sctx:                   sctx,
+		nullRejectCheckExprCtx: exprctx.WithNullRejectCheck(sctx.GetExprCtx()),
+	}
+}
+
+// GetNullRejectCheckExprCtx returns a context with null rejected check
+func (ctx *PlanCtxExtendedImpl) GetNullRejectCheckExprCtx() exprctx.ExprContext {
+	intest.AssertFunc(func() bool {
+		// assert `sctx.GetExprCtx()` should keep the same to avoid some unexpected behavior.
+		return ctx.nullRejectCheckExprCtx.ExprContext == ctx.sctx.GetExprCtx()
+	})
+	return ctx.nullRejectCheckExprCtx
```

Comment:
It is also feasible to just return:

```
func (ctx *PlanCtxExtendedImpl) GetNullRejectCheckExprCtx() exprctx.ExprContext {
	return exprctx.WithNullRejectCheck(sctx.GetExprCtx())
}
```

But we stored the context in `ctx. nullRejectCheckExprCtx` to reduce object allocations.